exciting_environments/classic_core_env.py

```diff
@@ -0,0 +1,308 @@
+        overall length of the simulation -> overall_time = actions.shape[0] * action_stepsize
+        The actions are interpolated with zero order hold inbetween their values.
+
+        Args:
+            init_state: The initial state of the simulation
+            actions: A set of actions to be applied to the environment, the value changes every
+            action_stepsize (shape=(n_action_steps, action_dim))
+            env_properties: The constant properties of the simulation
+            obs_stepsize: The sampling time for the observations
+            action_stepsize: The time between changes in the input/action
+        """
+
+        assert actions.ndim == 2, "The actions need to have two dimensions: (n_action_steps, action_dim)"
+        assert (
+            actions.shape[-1] == self.action_dim
+        ), f"The last dimension does not correspond to the action dim which is {self.action_dim}, but {actions.shape[-1]} is given"
+
+        init_physical_state_shape = jnp.array(tree_flatten(init_state.physical_state)[0]).T.shape
+        assert init_physical_state_shape == (self.physical_state_dim,), (
+            "The initial physical state needs to be of shape (env.physical_state_dim,) which is "
+            + f"{(self.physical_state_dim,)}, but {init_physical_state_shape} is given"
+        )
+
+        # denormalize actions
+        actions = actions * jnp.array(tree_flatten(env_properties.action_constraints)[0]).T
+
+        # compute states trajectory for given actions
+        states = self._ode_solver_simulate_ahead(
+            init_state, actions, env_properties.static_params, obs_stepsize, action_stepsize
+        )
+
+        # generate observations for all timesteps
+        observations = jax.vmap(self.generate_observation, in_axes=(0, None))(states, env_properties)
+
+        # delete first state because its initial state of simulation and not relevant for terminated
+        states_flatten, struct = tree_flatten(states)
+
+        states_without_init_state = tree_unflatten(struct, jnp.array(states_flatten)[:, 1:])
+
+        # reward = jax.vmap(self.generate_reward, in_axes=(0, 0, None))(
+        #     states_without_init_state,
+        #     jnp.expand_dims(jnp.repeat(actions, int(action_stepsize / obs_stepsize)), 1),
+        #     env_properties,
+        # )
+        # reward = 0
+
+        # generate truncated
+        # truncated = jax.vmap(self.generate_truncated, in_axes=(0, self.in_axes_env_properties))(
+        #     states, self.env_properties
+        # )
+
+        # generate terminated
+
+        # get last state so that the simulation can be continued from the end point
+        last_state = tree_unflatten(struct, jnp.array(states_flatten)[:, -1:])
+
+        # terminated = jax.vmap(self.generate_terminated, in_axes=(0, 0, self.in_axes_env_properties))(
+        #     states_without_init_state, reward, self.env_properties
+        # )
```

Comment:
please delete stuff like this. Commit the changes with a sensible commit message. If we decide to roll it back, you can look for such a commit message and undo it